src/app/dataset-view/additional-components/metadata-component/components/form-components/schema-field/schema-field.component.spec.ts

```diff
@@ -0,0 +1,160 @@
+
+describe("SchemaFieldComponent", () => {
+    let component: SchemaFieldComponent;
+    let fixture: ComponentFixture<SchemaFieldComponent>;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            declarations: [SchemaFieldComponent, TooltipIconComponent],
+            providers: [FormBuilder],
+            imports: [
+                ReactiveFormsModule,
+                MatIconModule,
+                MatTableModule,
+                NgbTypeaheadModule,
+                RxReactiveFormsModule,
+                NgbTooltipModule,
+            ],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(SchemaFieldComponent);
+        component = fixture.componentInstance;
+        component.form = new FormGroup({
+            schema: new FormArray([
+                new FormGroup({
+                    name: new FormControl("id", [
+                        RxwebValidators.unique(),
+                        RxwebValidators.required(),
+                        RxwebValidators.noneOf({
+                            matchValues: component.AVAILABLE_TYPES,
+                        }),
+                    ]),
+                    type: new FormControl("BIGINT", [
+                        RxwebValidators.required(),
+                    ]),
+                }),
+            ]),
+        });
+        component.controlName = "schema";
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+    });
+
+    it("should check add row to form array", () => {
+        const item: SchemaType = { name: "id", type: "BIGINT" };
+        component.focus$.next(item);
+        expect(component.items.length).toBe(1);
+        emitClickOnElement(fixture, '[data-test-id="add-row-button"]');
+        expect(component.items.length).toBe(2);
+    });
+
+    it("should check delete row to form array", () => {
+        expect(component.items.length).toBe(1);
+        emitClickOnElement(fixture, '[data-test-id="delete-row-button-0"]');
+        expect(component.items.length).toBe(0);
+    });
+
+    it("should check unique validation message", () => {
+        component.addRow();
+        fixture.detectChanges();
+        const controlName_1 = findElementByDataTestId(
+            fixture,
+            "name-control-1",
+        ) as HTMLInputElement;
+        controlName_1.value = "id";
+        controlName_1.dispatchEvent(new Event("input"));
+        fixture.detectChanges();
+        expect(component.nameControlError(1)).toBe("Name is not unique");
+    });
+
+    it("should check required validation message", () => {
+        component.addRow();
+        fixture.detectChanges();
+        const controlName_1 = findElementByDataTestId(
+            fixture,
+            "name-control-1",
+        ) as HTMLInputElement;
+        controlName_1.value = "";
+        controlName_1.dispatchEvent(new Event("input"));
```

Comment:
Please add another test helper for these dispatches - it is a typical test task. Please find if we have any similar ones in other tests, and refactor that.